app/views/products/show.html.erb

```diff
@@ -69,9 +69,11 @@
         <%= f.password_field :password, placeholder: 'Password' %>
         <%= f.check_box :remember_me %>
         <%= f.label :remember_me %>
-        <%= f.submit 'Sign in' %>
+        <%= f.hidden_field :redirect, value: "cart" %>
```

Comment:
@charliedayman no biggie, but lets remember to remove these hidden fields in the next pr.